### PR TITLE
MTDSA-4903 Added Auditing Boilerplate

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: app/v1/models/audit/GenericAuditDetail.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.audit
+
+import play.api.libs.json.{Json, OWrites}
+import v1.models.auth.UserDetails
+
+case class GenericAuditDetail(userType: String,
```
Comment (yangwong9664) [inline app/v1/models/audit/GenericAuditDetail.scala]: if userType can only be Individual or Agents, shouldn't this just be an enum/case object?

File: app/v1/models/audit/AuditEvent.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.audit
+
+case class AuditEvent[T](auditType: String, transactionName: String, detail: T)
```
Comment (yangwong9664) [inline app/v1/models/audit/AuditEvent.scala]: Why is "detail" a generic? If the detail fields (userType, agentReferenceNumber, nino, response, X-CorrelationId) are the same all the time, shouldn't we just make a normal case class?

Comment (yangwong9664) [inline app/v1/models/audit/AuditEvent.scala]: Where are the other fields? (auditSource, eventId, tags, generatedAt)

File: app/v1/models/audit/GenericAuditDetail.scala (hunk 1 above)
Comment (MarkAKelly) [reply app/v1/models/audit/GenericAuditDetail.scala]: Maybe, but as we dont _really_ worry too much about what  we are sending to Datastream we leave everything as raw as it comes in

File: app/v1/models/audit/AuditEvent.scala (hunk 2 above)
Comment (narendravijayarao) [reply app/v1/models/audit/AuditEvent.scala]: The naming convention I just followed as per our previous Audits, if you have any concerns better ask in the tech channel. And based on the team suggestions we can change

Comment (yangwong9664) [reply app/v1/models/audit/AuditEvent.scala]: The naming convention thing has been resolved by Mark, I'm talking about `detail: T` here, shouldn't this just be a normal case class if the detail field never changes? It always has (userType, agentReferenceNumber, nino, response, X-CorrelationId). Or is this not the case?

Comment (narendravijayarao) [reply app/v1/models/audit/AuditEvent.scala]: These are created at the creation of ExtendedDataEvent inside auditEvent method of AuditService

Comment (yangwong9664) [reply app/v1/models/audit/AuditEvent.scala]: Ok great

Comment (narendravijayarao) [reply app/v1/models/audit/AuditEvent.scala]: Yes, For Accounts there is no need to be Generic type... Before we have different audit types. The only reason I didn't change is we are following a template for all the services. So just used the same

**Pull request merged**